NitroxModel/Platforms/OS/Shared/ProcessEx.cs

```diff
@@ -102,21 +147,55 @@ public static ProcessEx GetFirstProcess(string procName, Func<ProcessEx, bool> p
                         }
                     }
                 }
-                catch (ArgumentException)
+                catch (Exception)
                 {
-                    // Process doesn't exist anymore, skip it
+                    // Process doesn't exist anymore or we don't have access, skip it
                     continue;
                 }
-                catch (UnauthorizedAccessException)
+            }
+        }
+        return null;
+    }
+
+    private static ProcessEx GetFirstProcessMacOS(string procName, Func<ProcessEx, bool> predicate, StringComparer comparer)
+    {
+        // Use process_listpids to get all PIDs
+        int[] pids = new int[1024];
+        int bytesNeeded = 0;
+        if (process_listpids(PROC_ALL_PIDS, 0, pids, pids.Length * sizeof(int), ref bytesNeeded) > 0)
+        {
+            int numProcesses = bytesNeeded / sizeof(int);
+            for (int i = 0; i < numProcesses; i++)
+            {
+                int pid = pids[i];
+                if (pid == 0) continue;
+
+                try
+                {
+                    ProcessEx processEx = new ProcessEx(pid);
+                    if (comparer.Compare(procName, processEx.Name) == 0)
+                    {
+                        if (predicate == null || predicate(processEx))
+                        {
+                            return processEx;
+                        }
+                    }
+                }
+                catch (Exception)
```

Comment:
@jumpers775 Hello there, just wanted to know the reason behind that change, why do we need our implementation for getting process by name ?
Which is already implemented by the .NET SDK with Process.GetProcessesByName, tried on Windows, Linux, MacOS and it works like a charm

Noticed that because I tried building a version from a mac, and was getting an error that process_listpids doesn't exist (https://github.com/apple-opensource/xnu/blob/24525736ba5b8a67ce3a8a017ced469abe101ad5/libsyscall/wrappers/libproc/libproc.c#L44)

Thanks 😄

Reply:
this PR was my introduction to c# so I just didn't know about that. my bad.